Framework/PythonInterface/test/python/plugins/algorithms/PeakMatchingTest.py

```diff
@@ -0,0 +1,313 @@
+        mock_get_default_peak_data.return_value = dict({
+            "Ag": {
+                "Z": 47,
+                "A": 107.87,
+                "Primary": {
+                    "K(4->1)": 3177.7,
+                    "L(4->2)": 900.7,
+                    "M(4->3)": 304.7,
+                    "6->5": 141
+                },
+                "Secondary": {
+                    "K(2->1)": 3140.6,
+                    "L(8->2)": 1347.8,
+                    "M(10->3)": 567,
+                    "8->6": 122.2
+                },
+                "Gammas": {
+                    "72Ge(n,n')72Ge": 691,
+                    "73Ge(n,g)74Ge": None,
+                    "74Ge(n,n')74Ge": 595.7
+                }
+            }
+        })
+
+        PeakMatching(PeakTable=self.input_table,
+                     AllPeaks = "rename_all",
+                     PrimaryPeaks= "rename_prim",
+                     SecondaryPeaks="rename_secon",
+                     SortedByEnergy="rename_sort",
+                     ElementLikelihood="rename_count")
+        prim = mtd['rename_prim']
+        print(prim.toDict())
+        secon = mtd['rename_secon']
+        all = mtd['rename_all']
+        sort = mtd['rename_sort']
+        count = mtd['rename_count']
+
+        correct_prim = {'Peak centre': [900.0, 306.0], 'Database Energy': [900.7, 304.7], 'Element': ['Ag', 'Ag'],
+                        'Transition': ['L(4->2)', 'M(4->3)'], 'Error': [0.8, 1.6],
+                        'Difference': [0.7, 1.3]}
+
+        correct_secon = {'Peak centre': [567.0], 'Database Energy': [567.0], 'Element': ['Ag'],
+                         'Transition': ['M(10->3)'], 'Error': [0.0], 'Difference': [0.0]}
+
+        correct_all = {'Peak centre': [567.0, 900.0, 306.0], 'Database Energy': [567.0, 900.7, 304.7],
+                       'Element': ['Ag', 'Ag', 'Ag'], 'Transition': ['M(10->3)', 'L(4->2)', 'M(4->3)'],
+                       'Error': [0.0, 0.8, 1.6], 'Difference': [0.0, 0.7, 1.3]}
+
+        correct_sort = {'Peak centre': [306.0, 567.0, 900.0], 'Database Energy': [304.7, 567.0, 900.7],
+                        'Element': ['Ag', 'Ag', 'Ag'], 'Transition': ['M(4->3)', 'M(10->3)', 'L(4->2)'],
+                        'Error': [1.6, 0.0, 0.8], 'Difference': [1.3, 0.0, 0.7]}
+
+        correct_count = {'Element': ['Ag'], 'Likelihood(arbitrary units)': [9]}
+
+        self.assertPeaksMatch(prim, correct_prim)
+        self.assertPeaksMatch(secon, correct_secon)
+        self.assertPeaksMatch(all, correct_all)
+        self.assertPeaksMatch(sort, correct_sort)
+        self.assertPeaksMatch(count, correct_count)
+
+        self.delete_if_present('rename_all')
+        self.delete_if_present('rename_prim')
+        self.delete_if_present('rename_secon')
+        self.delete_if_present('rename_sort')
+        self.delete_if_present('rename_count')
+
+if __name__ == '__main__':
+    unittest.main()
```

Comment:
Need a new line at the end of the file, and you also need to new lines between the class definition and the if statement. Should become something like:

```
            self.delete_if_present('rename_count')


if __name__ = '__main__':
   unittest.main()

```